### PR TITLE
Fix double proposal

### PR DESCRIPTION
**Pull request by GalRogozinski**
# Description

Makes sure that any new round change message that is processes after RC quorum is reached won't trigger sending a proposal

### CODE REVIEW

**Comment by GalRogozinski** [comment]
There are several ways to solve this:

1. Check it the PROPOSAL was committed the way it was done in PREPARE with COMMITs. 
https://github.com/bloxapp/ssv-spec/blob/2af3a73b3fe7a8f44ec154b71c67ca407e5d68bb/qbft/prepare.go#L28-L30.
But this checks inside the container... There could be a scenario where the COMMIT was broadcasted but another PREPARE was processed, resulting in 2 commits.
2. Flag once a proposal was broadcasted for a certain round and height. This will cover all scenarios but will farther complicate the state.
3. *The chosen solution*: ensure that only upon the first detection of RC quorum PROPOSAL will be broadcasted.

File: ssv/spectest/generate/state_comparison/tests_MsgProcessingSpecTest/qbft message processing valid justification.json [hunk 1]
```diff
@@ -0,0 +1,283 @@
+{
```
Comment (GalRogozinski) [inline ssv/spectest/generate/state_comparison/tests_MsgProcessingSpecTest/qbft message processing valid justification.json:1]: This file existed before

File: qbft/spectest/tests/roundchange/valid_justification_late.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package roundchange
+
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
+// ValidJustificationLate tests a valid rc quorum justification
```
Comment (alonmuroch) [inline qbft/spectest/tests/roundchange/valid_justification_late.go:10]: explanation is not complete, please write better detail

**Review by alonmuroch — CHANGES_REQUESTED** [review]

**Review by y0sher — APPROVED** [review]

File: qbft/spectest/tests/roundchange/valid_justification_late.go (hunk 2 above)
Comment (GalRogozinski) [reply qbft/spectest/tests/roundchange/valid_justification_late.go]: :+1:

**Review by alonmuroch — APPROVED** [review]

File: ssv/spectest/generate/state_comparison/tests_MsgProcessingSpecTest/qbft message processing valid justification.json (hunk 1 above)
Comment (nkryuchkov) [reply ssv/spectest/generate/state_comparison/tests_MsgProcessingSpecTest/qbft message processing valid justification.json:1]: @GalRogozinski 
This file was created under `ssv` directory, so the `qbft/.../qbft message processing valid justification.json` was unchanged. Did you intend to make changes to the `qbft/.../qbft message processing valid justification.json` and not to create this one (`ssv/.../qbft message processing valid justification.json`)?

**Pull request merged**